### PR TITLE
Merge farmer fix to 1.11

### PR DESCRIPTION
**Pull request by Raycoms**
Port the farmer fix to 1.11

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java [hunk 1]
```diff
@@ -125,14 +128,12 @@ private AIState startWorkingAtOwnBuilding()
     private AIState prepareForFarming()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:128]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java [hunk 2]
```diff
@@ -499,112 +366,162 @@ private boolean shouldHarvest(@NotNull final BlockPos position)
      * This (re)initializes a field.
      * Checks the block above to see if it is a plant, if so, breaks it. Then tills.
      */
-    private AIState initialize()
+    private AIState workAtField()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:369]: MAJOR Reduce the number of returns of this method 8, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 38 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 30 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L40](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L40): class "AbstractEntityAIBasic" has 56 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L205](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L205): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L533](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L533): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L983](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L983): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1032](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1032): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [Field.java#L98](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/Field.java#L98): Remove this unused "TAG_INITIALIZED" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/abff8e55d9cf645f77643f0c97b58ee9eb8b9c5a/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by xavierh** [comment]
@Raycoms this should be against testing/1.11 or we need to fix the Warehouse schematics independently for testing

**Comment by Raycoms** [comment]
@xavierh I put it into version but we can easily merge version into testing once again =)

**Comment by xavierh** [comment]
Would be nice to have the complete mesa style in release. What about the
fix for the farmer is it in testing/1.10?

**Comment by Raycoms** [comment]
It is

2017-06-16 10:35 GMT-03:00 xavierh <notifications@github.com>:

> Would be nice to have the complete mesa style in release. What about the
> fix for the farmer is it in testing/1.10?
>

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 38 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 30 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L40](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L40): class "AbstractEntityAIBasic" has 56 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L205](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L205): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L533](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L533): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L983](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L983): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1032](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1032): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [Field.java#L98](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/Field.java#L98): Remove this unused "TAG_INITIALIZED" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/540151e39a8b23b8c82670acc2c56f7406dfccc8/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
@OrionDevelopment @xavierh

**Comment by marvin-bitterlich** [comment]
LGTM

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1180/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 38 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 30 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L40](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L40): class "AbstractEntityAIBasic" has 56 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L205](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L205): Either log or rethrow this exception. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L533](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L533): Remove this unused private "requestWithoutSpam" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L983](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L983): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [AbstractEntityAIBasic.java#L1032](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java#L1032): Provide multiple methods instead of using "useItemDamage" to determine which action to take. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)
1. ![MAJOR][MAJOR] [Field.java#L98](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/Field.java#L98): Remove this unused "TAG_INITIALIZED" private field. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1068)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/4e66168079910c11b7727e1a6773afdb0b826943/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Pull request merged**